src/context/SWR.tsx

```diff
@@ -0,0 +1,81 @@
+import { SWRConfig } from 'swr';
+import { BaseComponentProps } from 'types';
+
+const middleware = {
+    cache: () => {
+        // When initializing, we restore the data from `localStorage` into a map.
+        const loggingMap = {
```

Comment:
Very basic cache (which swr's is also just a map) that puts a wrapper around the main functions and logs calls. These things get called A LOT - so that is why they are off by default.